agents/json_tool_gen.py

```diff
@@ -0,0 +1,162 @@
+"""
+Automated generation of function calling JSON payload for OpenAI
+using python type hints and a thin decorator
+
+Sean Browning
+"""
+from functools import update_wrapper
+import inspect
+from typing import (
+    List,
+    Union,
+    Callable,
+    Dict,
+    Any,
+    Literal,
+    get_type_hints,
+    get_origin,
+    get_args,
+)
+from types import NoneType
+
+PYTHON_TO_OAI_SCHEMA = {
+    str: "string",
+    int: "integer",
+    float: "number",
+    bool: "boolean",
+    List: "array",
+    Dict: "object",
+    NoneType: "null",
+}
+
+class AgentCallable:
```

Comment:
Was a decent thought, but writing as a class creates more hassles than it solves, I think. I'm going to re-write purely functional.